projects/ad9361/src/ad9361.c

```diff
@@ -572,35 +572,29 @@ const char *ad9361_ensm_states[] = {
  * @param num The number of bytes to read.
  * @return 0 in case of success, negative error code otherwise.
  */
-int32_t ad9361_spi_readm(struct spi_device *spi, uint32_t reg,
+int32_t ad9361_spi_readm(struct spi_desc *spi, uint32_t reg,
 			 uint8_t *rbuf, uint32_t num)
 {
-	uint8_t buf[2];
-	int32_t ret;
+	int32_t ret = 0;
 	uint16_t cmd;
-
+	uint8_t *rbuffer;
 	if (num > MAX_MBYTE_SPI)
 		return -EINVAL;
 
 	cmd = AD_READ | AD_CNT(num) | AD_ADDR(reg);
-	buf[0] = cmd >> 8;
-	buf[1] = cmd & 0xFF;
+	rbuffer = (uint8_t *) malloc(num + 2);
+	rbuffer[0] = cmd >> 8;
+	rbuffer[1] = cmd & 0xFF;
+	ret = spi_write_and_read(spi, &rbuffer[0], 2 + num);
 
-	ret = spi_write_then_read(spi, &buf[0], 2, rbuf, num);
 	if (ret < 0) {
 		dev_err(&spi->dev, "Read Error %"PRId32, ret);
-		return ret;
-	}
-#ifdef _DEBUG
-	{
-		int32_t i;
-		for (i = 0; i < num; i++)
-			dev_dbg(&spi->dev, "%s: reg 0x%"PRIX32" val 0x%X",
-				__func__, reg--, rbuf[i]);
+	} else {
```

Comment:
nitpick: now that the if and the else both only guard one statement there should be no need for the brackets.